modules/EvseV2G/connection.cpp

```diff
@@ -985,3 +1021,51 @@ int connection_start_servers(struct v2g_context* ctx) {
 
     return 0;
 }
+int create_udp_socket(const int udp_port) {
+    // find suitable addresses
+    addrinfo hints;
+    addrinfo *result, *result_pointer;
+    int udp_socket;
+
+    memset(&hints, 0, sizeof(addrinfo));
+    hints.ai_family = AF_UNSPEC;    // allow IPv4 or IPv6
+    hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+    hints.ai_flags = AI_ALL;        // For IPv4 or IPv6
+    hints.ai_protocol = 0;          // OS selects best protocol (best practise)
+
+    // convert port to string
+    char port_str[6];
+    snprintf(port_str, sizeof(port_str), "%d", udp_port);
+
+    //              alternatively:
+    //              "localhost"
+    auto get_addrinfo_result = getaddrinfo(nullptr, port_str, &hints, &result);
+
+    if (get_addrinfo_result != 0) {
+        dlog(DLOG_LEVEL_ERROR, "getaddrinfo: %s", gai_strerror(get_addrinfo_result));
+        return -1;
+    }
+
+    // create the socket
+    for (result_pointer = result; result_pointer != nullptr; result_pointer = result_pointer->ai_next) {
+        udp_socket = socket(result_pointer->ai_family, result_pointer->ai_socktype, result_pointer->ai_protocol);
```

Comment:
what is the idea behind this?  In case creating an udp socket fails, just try the next available addresses?  This doesn't really look deterministic to me.

Reply:
I'm currently discussing this with @MarzellT, whether this is too much and could be done more simply.

Reply:
It might make sense to simplify this. I thought this was the "correct" way of creating sockets but I must admit I'm not sure. Basically, you would let the OS take care of creating an appropriate socket given the `hints`. It would be possible to request IPv6 explicitly changing the `ai_family` type in the hints but I'm not sure if the whole process is necessary. I don't really think it's necessary to check more than the first result of `getaddrinfo` but at the same time I'm not sure what could be the possible results of that function call